File: 프로그래머스/1/12915. 문자열 내 마음대로 정렬하기/문자열 내 마음대로 정렬하기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<string> solution(vector<string> strings, int n) {
    auto func = ([&n](string a, string b){
        if(a[n] == b[n]) return a < b;        
        return a[n] < b[n];
    });
    sort(strings.begin(), strings.end(), func);
        
    return strings;
}

File: 프로그래머스/1/12915. 문자열 내 마음대로 정렬하기/문자열 내 마음대로 정렬하기.cs
using System;

public class Solution {
    public string[] solution(string[] strings, int n) {        
        Array.Sort(strings, (a,b)=>
        {                        
            int cmp = a[n].CompareTo(b[n]);
            if(cmp == 0) cmp = a.CompareTo(b);
            return cmp;
        });
        
        
        return strings;
    }
}